architxt/database/__init__.py

```diff
@@ -0,0 +1,240 @@
+from sqlalchemy import create_engine, inspect, text
+from sqlalchemy.exc import SQLAlchemyError
+from sqlalchemy.orm import sessionmaker
+
+from architxt.tree import NodeLabel, NodeType, Tree
+
+
+def read_database(db_path: str, db_type: str) -> None:
+    """
+    Read the database, retrieve table information, process table relations, and print the results in a tree format.
+
+    :param db_path: Path to the database.
+    :param db_type: Type of the database.
+    """
+    session = None
+    try:
+        # Create a connection to the SQLite database
+        engine = create_engine(f'{db_type}:///{db_path}')
```

Comment:
Why not accept the full database URL as argument instead of multiple parameters? This would make the code more flexible and better support features like authentication.